src/modules/colorPicker/ColorPickerUI/ViewModels/ColorEditorViewModel.cs

```diff
@@ -133,100 +148,184 @@ private void ColorsHistory_CollectionChanged(object sender, System.Collections.S
             }
         }
 
-        private void DeleteSelectedColor()
+        private void DeleteSelectedColors(object selectedColors)
         {
-            // select new color on the same index if possible, otherwise the last one
-            var indexToSelect = SelectedColorIndex == ColorsHistory.Count - 1 ? ColorsHistory.Count - 2 : SelectedColorIndex;
-            ColorsHistory.RemoveAt(SelectedColorIndex);
-            SelectedColorIndex = indexToSelect;
+            var colorsToRemove = ((IList)selectedColors).OfType<Color>().ToList();
+            var indicesToRemove = colorsToRemove.Select(color => ColorsHistory.IndexOf(color)).ToList();
+
+            foreach (var color in colorsToRemove)
+            {
+                ColorsHistory.Remove(color);
+            }
+
+            SelectedColorIndex = ComputeWhichIndexToSelectAfterDeletion(colorsToRemove.Count + ColorsHistory.Count, indicesToRemove);
             SessionEventHelper.Event.EditorHistoryColorRemoved = true;
         }
 
+        private void ExportSelectedColorsByColor(object selectedColors)
+        {
+            ExportColors(selectedColors, GroupExportedColorsBy.Color);
+        }
+
+        private void ExportSelectedColorsByFormat(object selectedColors)
+        {
+            ExportColors(selectedColors, GroupExportedColorsBy.Format);
+        }
+
+        private void ExportColors(object colorsToExport, GroupExportedColorsBy method)
+        {
+            var colors = SerializationHelper.ConvertToDesiredColorFormats((IList)colorsToExport, ColorRepresentations, method);
+
+            var dialog = new SaveFileDialog
+                  {
+                      Title = "Save selected colors to",
+                      Filter = "Text Files (*.txt)|*.txt|Json Files (*.json)|*.json",
+                  };
+
+            if (dialog.ShowDialog() == true)
+            {
+                var extension = Path.GetExtension(dialog.FileName);
+
+                var contentToWrite = extension.ToUpperInvariant() switch
+                {
+                    ".TXT" => colors.ToTxt(';'),
```

Comment:
isn't this the same as csv?

Reply:
Yes. But I don't get your question. Could you please elaborate on what needs to be changed here?

Reply:
The question is if we should allow this with csv firmst too?